src/BuildOpt.h

```diff
@@ -1040,14 +1040,37 @@
 
 #if defined(RADIOLIB_DEBUG)
   #if defined(RADIOLIB_BUILD_ARDUINO)
-    #define RADIOLIB_DEBUG_PRINT(...) { RADIOLIB_DEBUG_PORT.print(__VA_ARGS__); }
-    #define RADIOLIB_DEBUG_PRINTLN(...) { RADIOLIB_DEBUG_PORT.println(__VA_ARGS__); }
+// https://github.com/esp8266/Arduino/blob/65579d29081cb8501e4d7f786747bf12e7b37da2/cores/esp8266/Print.cpp#L50
+size_t serialPrintf(const char *format, ...) {
+    va_list arg;
+    va_start(arg, format);
+    char temp[64];
+    char* buffer = temp;
+    size_t len = vsnprintf(temp, sizeof(temp), format, arg);
+    va_end(arg);
+    if (len > sizeof(temp) - 1) {
+        buffer = new char[len + 1];
+        if (!buffer) {
+            return 0;
+        }
+        va_start(arg, format);
+        vsnprintf(buffer, len + 1, format, arg);
+        va_end(arg);
+    }
+    len = RADIOLIB_DEBUG_PORT.write((const uint8_t*) buffer, len);
+    if (buffer != temp) {
+        delete[] buffer;
+    }
+    return len;
+}
+    #define RADIOLIB_DEBUG_PRINT(...) serialPrintf(__VA_ARGS__)
```

Comment:
> If a generic printf() function will be added, then it will be better to have it as a separate macro RADIOLIB_DEBUG_PRINTF instead of changing the behavior of the currently implemented PRINT/PRINTLN.

My idea was to fix the difference of the behaviour between Arduino's `RADIOLIB_DEBUG_PRINT` and the fallback's `RADIOLIB_DEBUG_PRINT`, Serial.print does not support format like sprintf, and sprintf can't takes other argument than a `const char`.

If we create a `RADIOLIB_DEBUG_PRINTF`, what PRINT/PRINTLN will become? And how we can ensure that they have the same behaviour no matter the build platform?

Reply:
Hmm, you're right. In that case let's keep this new implementation.